src/instr.cpp

```diff
@@ -1025,19 +1025,13 @@ bool checkInstruction(Instruction* ins, Function* F, RewriterConfig rw_config, i
  * @return size of global variable.
  */
 uint64_t getGlobalVarSize(GlobalVariable* GV, const Module& M) {
-
-    DataLayout* DL = new DataLayout(&M);
-
     Type* Ty = GV->getType()->getElementType();
 
     if (!Ty->isSized())
         return 0;
 
-    uint64_t size = DL->getTypeAllocSize(Ty);
-
-    delete DL;
-
-    return size;
+    std::unique_ptr<DataLayout> DL(new DataLayout(&M));
```

Comment:
Actually, creating a new DataLayout was the old-LLVM way. Now, llvm::Module provides a getter for its internal DataLayout, so it would be better to use that: `M.getDataLayout()`. Could you go this way, please?

Reply:
Sure, I have force-pushed the proposed changes.